.travis/install.sh

```diff
@@ -1,121 +1,200 @@
+        else
+            git clone https://github.com/colesbury/ccache -b ccbin "$BUILD_CCACHE_DIR"
+            pushd "$BUILD_CCACHE_DIR"
+            ./autogen.sh
+            ./configure
+            make "-j$(nproc)"
+            popd
+        fi
+
+        # Overwrite ccache symlink wrappers
+        pushd /usr/local/bin
+        sudo ln -sf "${BUILD_CCACHE_DIR}/ccache" gcc
+        sudo ln -sf "${BUILD_CCACHE_DIR}/ccache" g++
+        sudo ln -sf "${BUILD_CCACHE_DIR}/ccache" nvcc
+        popd
+
+        #################
+        # Install CMake #
+        #################
+        # Newer version required to get cmake+ccache+nvcc to work
+        _cmake_installer=/tmp/cmake.sh
+        wget -O "$_cmake_installer" https://cmake.org/files/v3.8/cmake-3.8.2-Linux-x86_64.sh
+        sudo bash "$_cmake_installer" --prefix=/usr/local --skip-license
+        rm -rf "$_cmake_installer"
+
+        ################
+        # Install CUDA #
+        ################
+        CUDA_REPO_PKG='cuda-repo-ubuntu1404_8.0.44-1_amd64.deb'
+        CUDA_PKG_VERSION='8-0'
+        CUDA_VERSION='8.0'
+        wget "https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1404/x86_64/${CUDA_REPO_PKG}"
+        sudo dpkg -i "$CUDA_REPO_PKG"
+        rm -f "$CUDA_REPO_PKG"
+        sudo apt-get update
+        $APT_INSTALL_CMD \
+            "cuda-core-${CUDA_PKG_VERSION}" \
+            "cuda-cublas-dev-${CUDA_PKG_VERSION}" \
+            "cuda-cudart-dev-${CUDA_PKG_VERSION}" \
+            "cuda-curand-dev-${CUDA_PKG_VERSION}" \
+            "cuda-driver-dev-${CUDA_PKG_VERSION}" \
+            "cuda-nvrtc-dev-${CUDA_PKG_VERSION}"
+        # Manually create CUDA symlink
+        sudo ln -sf /usr/local/cuda-$CUDA_VERSION /usr/local/cuda
+
+        #################
+        # Install cuDNN #
+        #################
+        CUDNN_REPO_PKG='nvidia-machine-learning-repo-ubuntu1404_4.0-2_amd64.deb'
+        CUDNN_PKG_VERSION='6.0.20-1+cuda8.0'
+        wget "https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1404/x86_64/${CUDNN_REPO_PKG}"
+        sudo dpkg -i "$CUDNN_REPO_PKG"
+        rm -f "$CUDNN_REPO_PKG"
+        sudo apt-get update
+        $APT_INSTALL_CMD \
+            "libcudnn6=${CUDNN_PKG_VERSION}" \
+            "libcudnn6-dev=${CUDNN_PKG_VERSION}"
+    fi
+
+    if [ "$BUILD_MKL" = 'true' ]; then
+        ###############
+        # Install MKL #
+        ###############
+        _mkl_key=/tmp/mkl.pub
+        wget -O "$_mkl_key" http://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2019.PUB
+        sudo apt-key add "$_mkl_key"
+        rm -f "$_mkl_key"
+        echo 'deb http://apt.repos.intel.com/mkl all main' | sudo tee /etc/apt/sources.list.d/intel-mkl.list
+        sudo apt-get update
+        $APT_INSTALL_CMD intel-mkl-64bit-2017.3-056
+    fi
+
+    #################
+    # Install ninja #
+    #################
+    # NNPACK needs a recent version
```

Comment:
Why isn't NNPACK treated like MKL (it's dependencies are being built for cuda but the cuda build specifies -DUSE_NNPACK=OFF)?